scripts/process-csv-images

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# This script processes a given cluster service version file (csv) to replace all occurrences of internal image registries with a deloeran version and generate an image_mirror_mapping file.
+#
+# Example: Running the script and checking the changes produced
+#
+#$ ./process-image-manifests registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-3scale-rhel7-operator-metadata:1.11.0-2
+#$ grep delorean manifests/integreatly-3scale/3scale-0.5.0/3scale-operator.v0.5.0.clusterserviceversion.yaml
+#    image: quay.io/integreatly/delorean/3scale-amp2-apicast-gateway-rhel8_3scale2.8
+#    image: quay.io/integreatly/delorean/3scale-amp2-backend-rhel7_3scale2.8
+#    image: quay.io/integreatly/delorean/3scale-amp2-system-rhel7_3scale2.8
+#    image: quay.io/integreatly/delorean/3scale-amp2-zync-rhel7_3scale2.8
+#    image: quay.io/integreatly/delorean/3scale-amp2-memcached-rhel7_3scale2.8
+#$ cat manifests/integreatly-3scale/image_mirror_mapping
+#registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-apicast-gateway-rhel8:3scale2.8 quay.io/integreatly/delorean/3scale-amp2-apicast-gateway-rhel8_3scale2.8
+#registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-backend-rhel7:3scale2.8 quay.io/integreatly/delorean/3scale-amp2-backend-rhel7_3scale2.8
+#registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-system-rhel7:3scale2.8 quay.io/integreatly/delorean/3scale-amp2-system-rhel7_3scale2.8
+#registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-zync-rhel7:3scale2.8 quay.io/integreatly/delorean/3scale-amp2-zync-rhel7_3scale2.8
+#registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-memcached-rhel7:3scale2.8 quay.io/integreatly/delorean/3scale-amp2-memcached-rhel7_3scale2.8
+
+set -e
+
+WORK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+MANIFESTS_DIR=${WORK_DIR}/../manifests
+MANIFEST_TMP_DIR=${MANIFESTS_DIR}/tmp
+
+RH_REGISTRY_PROD=registry.redhat.io
+RH_REGISTRY_STAGE=registry.stage.redhat.io
+RH_REGISTRY_OSBS=registry-proxy.engineering.redhat.com/rh-osbs
+DELOREAN_REGISTRY=quay.io/integreatly/delorean
+
+collect_csv_images() {
+    csv_images=(`grep -o $RH_REGISTRY_STAGE.* $1`)
+}
+
+image_exists() {
+    echo "Checking if image exists $1"
+    #ToDo Need a way to determine if an image exists in any registry
+    #skopeo inspect docker://$1 > /dev/null
+    return 1
+}
+
+process_csv_images() {
+    echo "Generate Image Mirror Mapping for $2"
+    image_mirror_mapping=${MANIFESTS_DIR}/integreatly-${2}/image_mirror_mapping
+
+    for image in "${csv_images[@]}"
+    do
+        if ! image_exists $image; then
```

Comment:
@mikenairn I am not sure I get this part. Why do we need to check if the staging image exists anywhere? Would it be enough if we check if the image_mirror_mapping file has the corresponding entry?

Reply:
I've removed this for the moment as the logic here was slightly wrong since all stage images also need to be dealt with even if they can be accessed from the staging registry.

We will eventually need to be able to determine whether or not an image is actually GAd or not, and potentially have to change the registry to the production one since the manifests never seem to get updated currently.